ui/pages/onboarding-flow/onboarding-flow.js

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useState } from 'react';
+import { Switch, Route, useHistory } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import Unlock from '../unlock-page';
+import {
+  ONBOARDING_CREATE_PASSWORD_ROUTE,
+  ONBOARDING_REVIEW_SRP_ROUTE,
+  ONBOARDING_CONFIRM_SRP_ROUTE,
+  ONBOARDING_UNLOCK_ROUTE,
+  DEFAULT_ROUTE,
+} from '../../helpers/constants/routes';
+import {
+  getCompletedOnboarding,
+  getFirstTimeFlowTypeRoute,
+  getIsInitialized,
+  getIsUnlocked,
+  getSeedPhraseBackedUp,
+} from '../../selectors';
+import {
+  createNewVaultAndGetSeedPhrase,
+  unlockAndGetSeedPhrase,
+} from '../../store/actions';
+import OnboardingFlowSwitch from './onboarding-flow-switch/onboarding-flow-switch';
+import NewAccount from './new-account/new-account';
+import ReviewRecoveryPhrase from './recovery-phrase/review-recovery-phrase';
+import ConfirmRecoveryPhrase from './recovery-phrase/confirm-recovery-phrase';
+
+export default function OnboardingFlow() {
+  const [seedPhrase, setSeedPhrase] = useState('');
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const isInitialized = useSelector(getIsInitialized);
+  const isUnlocked = useSelector(getIsUnlocked);
+  const completedOnboarding = useSelector(getCompletedOnboarding);
+  const seedPhraseBackedUp = useSelector(getSeedPhraseBackedUp);
+  const nextRoute = useSelector(getFirstTimeFlowTypeRoute);
+
+  useEffect(() => {
+    if (process.env.ONBOARDING_V2) {
+      history.push(ONBOARDING_CREATE_PASSWORD_ROUTE);
+      return;
+    }
+
+    if (completedOnboarding && seedPhraseBackedUp) {
+      history.push(DEFAULT_ROUTE);
+      return;
+    }
+
+    if (isInitialized && !isUnlocked) {
+      history.push(ONBOARDING_UNLOCK_ROUTE);
+    }
+  }, [
+    history,
+    completedOnboarding,
+    isInitialized,
+    isUnlocked,
+    seedPhraseBackedUp,
+  ]);
+
+  const handleCreateNewAccount = async (password) => {
+    try {
+      const newSeedPhrase = await dispatch(
+        createNewVaultAndGetSeedPhrase(password),
+      );
+      setSeedPhrase(newSeedPhrase);
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  };
+
+  const handleUnlock = async (password) => {
+    try {
+      const retreivedSeedPhrase = await dispatch(
+        unlockAndGetSeedPhrase(password),
+      );
+      setSeedPhrase(retreivedSeedPhrase);
+      history.push(nextRoute);
+    } catch (error) {
+      throw new Error(error.message);
```

Comment:
What is the purpose of this catch block, and the one above on line 67? These would have the effect of stripping the error of any non-message properties, like `code` and `stack`.

Reply:
Honestly this is a copy from current [first-time-flow.component.js file](https://github.com/MetaMask/metamask-extension/blob/6187ab9b01ea757366a42d22b7292413c0f2b119/ui/pages/first-time-flow/first-time-flow.component.js#L97). I created a new file to convert to a functional component and keep the use of env variables down. But you're totally right this is silly. I will remove the try/catch here.